### PR TITLE
Refactor ledger#entry()

### PR DESCRIPTION
**Pull request by andschwa**
The entry front-end was buggy because of how it escaped the query. It
was fixed by refactoring to use `systemlist(s:ledger_cmd())` instead of
`read !` with `shellescape()`.

If the call to `ledger entry` fails, it echoes the failure message, and
leaves the input query to be amended by the user.

The refactor follows what seems to be the current pattern of the codebase.

### CODE REVIEW

**Comment by andschwa** [comment]
@kljohann is this repository still maintained?

File: autoload/ledger.vim [hunk 1]
```diff
@@ -490,6 +482,20 @@ function! s:ledger_cmd(file, args)
...
+" Use current line as input to ledger entry and replace with output. If there
+" are errors, they are echoed instead.
+func! ledger#entry()
+  let l:output = systemlist(s:ledger_cmd(g:ledger_main, join(["entry", getline('.')])))
+  " Errors may occur
+  if v:shell_error
+    echomsg join(l:output)
+    return
+  endif
+  " Append output so we insert instead of overwrite, then delete line
```
Comment (kljohann) [inline autoload/ledger.vim]: Can you add code to filter out warning lines here?
```viml
  " Filter out warnings
  let l:output = filter(l:output, "v:val !~? '^Warning: '")
```

**Comment by kljohann** [comment]
Yes, I just have been short on time, sorry. Thanks for your contribution. :+1:

File: autoload/ledger.vim (hunk 1 above)
Comment (andschwa) [inline autoload/ledger.vim]: And done!

**Comment by andschwa** [comment]
No problem! I understand that problem all too well. Change implemented and I've been updating my books :ledger: using this code, so far so good.

**Comment by kljohann** [comment]
Thanks! :)

**Pull request merged**